bundles/aliyun/src/main/java/org/apache/gravitino/oss/fs/OSSCredentialProvider.java

```diff
@@ -0,0 +1,154 @@
+    this.configuration = conf;
+  }
+
+  @Override
+  public void setCredentials(Credentials credentials) {}
+
+  @Override
+  public Credentials getCredentials() {
+    // If the credentials are null or about to expire, refresh the credentials.
+    if (basicCredentials == null || System.currentTimeMillis() >= expirationTime) {
+      synchronized (this) {
+        refresh();
+      }
+    }
+
+    return basicCredentials;
+  }
+
+  private void refresh() {
+    String[] idents = filesetIdentifier.split("\\.");
+    String catalog = idents[1];
+
+    FilesetCatalog filesetCatalog = client.loadCatalog(catalog).asFilesetCatalog();
+
+    Fileset fileset = filesetCatalog.loadFileset(NameIdentifier.of(idents[2], idents[3]));
+    Credential[] credentials = fileset.supportsCredentials().getCredentials();
+    Optional<Credential> optionalCredential = getCredential(credentials);
+
+    if (!optionalCredential.isPresent()) {
+      LOG.warn("No credential found for fileset: {}, try to use static AKSK", filesetIdentifier);
+      expirationTime = Long.MAX_VALUE;
+      this.basicCredentials =
+          new DefaultCredentials(
+              configuration.get(Constants.ACCESS_KEY_ID),
+              configuration.get(Constants.ACCESS_KEY_SECRET));
+      return;
+    }
+
+    Credential credential = optionalCredential.get();
+    Map<String, String> credentialMap = credential.toProperties();
+
+    String accessKeyId = credentialMap.get(GRAVITINO_OSS_SESSION_ACCESS_KEY_ID);
+    String secretAccessKey = credentialMap.get(GRAVITINO_OSS_SESSION_SECRET_ACCESS_KEY);
+
+    if (OSSTokenCredential.OSS_TOKEN_CREDENTIAL_TYPE.equals(
+        credentialMap.get(Credential.CREDENTIAL_TYPE))) {
+      String sessionToken = credentialMap.get(GRAVITINO_OSS_TOKEN);
+      basicCredentials = new BasicCredentials(accessKeyId, secretAccessKey, sessionToken);
+    } else {
+      basicCredentials = new DefaultCredentials(accessKeyId, secretAccessKey);
+    }
+
+    if (credential.expireTimeInMs() > 0) {
+      expirationTime =
+          System.currentTimeMillis()
+              + (long)
+                  ((credential.expireTimeInMs() - System.currentTimeMillis())
+                      * EXPIRATION_TIME_FACTOR);
+    }
+  }
+
+  /**
+   * Get the credential from the credential array. Using dynamic credential first, if not found,
+   * uses static credential.
+   *
+   * @param credentials The credential array.
+   * @return An optional credential.
+   */
+  private Optional<Credential> getCredential(Credential[] credentials) {
+    // Use dynamic credential if found.
+    Optional<Credential> dynamicCredential =
+        Arrays.stream(credentials)
+            .filter(
+                credential ->
+                    credential
+                        .credentialType()
+                        .equals(OSSTokenCredential.OSS_TOKEN_CREDENTIAL_TYPE))
+            .findFirst();
+    if (dynamicCredential.isPresent()) {
+      return dynamicCredential;
+    }
```

Comment:
If you use `Optional`, please monad style (functional programming), not using `if...`, which is not elegant.

Reply:
OK, let me make some adjustments.